apps/researcher/src/app/[locale]/objects/[id]/(provenance)/buttons.tsx

```diff
@@ -0,0 +1,73 @@
+'use client';
+
+import classNames from 'classnames';
+import {useProvenance} from './provenance-store';
+import {ButtonHTMLAttributes, ReactNode} from 'react';
+import {useTranslations} from 'next-intl';
+
+interface SelectEventsButtonProps
+  extends ButtonHTMLAttributes<HTMLButtonElement> {
+  ids: string[];
+  children: ReactNode;
+}
+
+export function SelectEventsButton({ids, children}: SelectEventsButtonProps) {
+  const {setSelectedEvents, selectedEvents} = useProvenance();
+  const selected = ids.some((id: string) => selectedEvents.includes(id));
+
+  const handleClick = () => {
+    if (selected) {
+      setSelectedEvents(selectedEvents.filter(id => !ids.includes(id)));
+    } else {
+      setSelectedEvents(ids);
+    }
+  };
+
+  return (
+    <button
+      className={classNames(
+        'bg-consortiumBlue-600 rounded-full h-8 min-w-[33px] px-1 flex justify-center items-center border-2 transition text-xs font-medium whitespace-nowrap',
+        {
+          'hover:bg-consortiumBlue-700 border-white hover:border-consortiumGreen-300 hover:text-consortiumGreen-300':
+            !selected,
+          'bg-consortiumBlue-700 border-consortiumGreen-300 text-consortiumGreen-300':
+            selected,
+        }
+      )}
+      onClick={handleClick}
+    >
+      {children}
+    </button>
+  );
+}
+
+export function ToggleViewButtons() {
+  const t = useTranslations('Provenance');
+  const {showTimeline, showDataTable, setShowTimeline, setShowDataTable} =
+    useProvenance();
+
+  const toggleTimeline = () => {
```

Comment:
Would the shorter version also work? `const toggleTimeline = () => setShowTimeline(!showTimeline);`